### ISSUE TITLE
Configure Homu as a first-class GitHub application

### ISSUE THREAD

**Issue by aneeshusa** [servo/saltfs#283]
Currently, we have one shared webhook that must be manually handed out and configured to add a new repo, and we have to update the Salt states.

At the very least, we should be generating per-repo tokens so each repo gets a different webhook. Ideally, we would register Homu as a Github application so this is automated and doesn't require sending a PR to this repo to add a new repo.

This will likely get set up at `homu.servo.org`.

See https://github.com/servo/saltfs/pull/279#issue-143694028. cc @larsbergstrom 

I don't know that much about Homu, but based on the FAQ at http://homu.io/, this would seem to entail Homu maintaining its own database (PostgreSQL? I think we can get away with SQLite for now) of repos. If so, we should have a backup + restore story for this database.

**Comment by Manishearth** [comment]
Yeah, I've wanted to do this for a while. In theory it shouldn't be too hard. Homu already maintains its own database, the issue is that the database isn't required to be persistent. So deleting the database works fine (usually), and is sometimes something I've done (with backup) to bring homu back to a consistent state. This also means that if servo-master gets moved, homu would have to sync all the apps all over again. If we do add a persistent db, there should be a way for salt or something to keep the db backed up somewhere, at least the relevant tables.
